[GSoC2024] fix and refactor import-dataset-modal code (#7428) (#7599)
opencv/cvat: changelog.d/20240313_014112_umangapatel123_refactor_into_usereducer.md

### Fixed

> - Incorrect file name usage in importing annotation via cloud storage

Reviewer: ```suggestion
- Incorrect file name usage when importing annotations from a cloud storage
```

> - set `file` and `fileName` to null after import is done and change `useState` to `useReducer`

Reviewer: We should describe here what it means for a CVAT user, not for developer.

Author: can you please review now?
